### ISSUE TITLE
Changing PlotStyle in a loop can raise matplotlib latex errors

### ISSUE THREAD

**Issue by prismv** [Axect/Peroxide_Gallery#3]
See MR #1 for code that creates this bug - as discussed in comments. 

MR #2 should provide a working example based on improved upstream functions.

**Comment by Axect** [comment]
Hey there, thanks for jumping on this so quickly! I really appreciate your enthusiasm. I just wanted to clear up a couple of things:

1. The bug we found is actually in the Peroxide library itself, not in Peroxide_Gallery. When I mentioned opening an issue, I meant in the main Peroxide repo. That way, we can tackle the root of the problem.

2. Would you mind if I move this issue over to the Peroxide repo? It'll help us keep track of things better.

3. Once we fix this in Peroxide, we can circle back to your PR in Peroxide_Gallery. We should be able to uncomment your code and get it working smoothly.

4. It's great that you've already submitted a PR to Peroxide_Gallery - nice initiative! We'll just need to put it on hold until we sort out the Peroxide issue. After that, we can review it and hopefully merge it in with any tweaks needed.

What do you think about handling it this way? I think it'll let us solve the main issue first, then bring your contribution into Peroxide_Gallery smoothly.

If you're okay with this, I can move the issue to the Peroxide repo now. Let me know what you think.

**Comment by prismv** [comment]
No, I don't mind if you shift this issue to the library repro. All statements are fine for me.
